eth-bytecode-db/eth-bytecode-db/src/verification/db.rs

```diff
@@ -131,21 +111,116 @@ async fn insert_source_files(
                 .context("insert into \"files\"")?,
             }
         };
+        result.push(file);
+    }
+
+    Ok(result)
+}
+
+async fn insert_source_details(
+    txn: &DatabaseTransaction,
+    source: types::Source,
+    file_models: &[files::Model],
+) -> Result<(sources::Model, bool), anyhow::Error> {
+    let abi = match source.abi {
+        None => None,
+        Some(abi) => serde_json::from_str(&abi).context("deserialize abi")?,
```

Comment:
That's a .map

Reply:
We cannot apply error propagation (`?`) in the `.map` closure

Reply:
Didn't know about `.transpose()`. Will refactor it, thanks